arroyo-worker/src/operators/sources/kafka/mod.rs

```diff
@@ -164,74 +164,111 @@ where
     }
 
     async fn run(&mut self, ctx: &mut Context<(), T>) -> SourceFinishType {
-        let consumer = self.get_consumer(ctx).await.unwrap();
-
-        let rate_limiter = RateLimiter::direct(Quota::per_second(self.messages_per_second));
-        let mut offsets = HashMap::new();
-        loop {
-            select! {
-                message = consumer.recv() => {
-                    match message {
-                        Ok(msg) => {
-                            if let Some(v) = msg.payload() {
-                                ctx.collector.collect(Record {
-                                    timestamp: from_millis(msg.timestamp().to_millis().unwrap() as u64),
-                                    key: None,
-                                    value: self.serialization_mode.deserialize_slice(&v).unwrap(),
-                                }).await;
-                                offsets.insert(msg.partition(), msg.offset());
-                                rate_limiter.until_ready().await;
+        let consumer = self.get_consumer(ctx).await;
+        match consumer {
```

Comment:
@mwylde is this the way you had in mind to handle the error? It seems not so nice to have such a huge match statement but I'm not sure what the alternative is.

Reply:
Yeah, I wouldn't put everything in a giant match statement, since the error handling is something that's just relevant at the beginning. Instead I would put just the assignment in the match:

```rust
let consumer = match self.get_consumer(ctx).await {
  Ok(consumer) => consumer,
  Err(err) => {
        ctx.control_tx
            .send(ControlResp::Error {
                operator_id: ctx.task_info.operator_id.clone(),
                task_index: ctx.task_info.task_index.clone(),
                message: "Error while getting kafka consumer".to_string(),
                details: format!("{:?}", err),
            })
            .await
            .unwrap();
        panic!("Error while getting kafka consumer")
  }
```